coord/src/rmr/redise_parser/grammar.c

```diff
@@ -1006,15 +1006,15 @@ static void yy_reduce(
       case 9: /* shardid ::= STRING */
 #line 145 "grammar.y"
 {
-	yylhsminor.yy1 = strdup(yymsp[0].minor.yy0.strval);
+	yylhsminor.yy1 = rm_strdup(yymsp[0].minor.yy0.strval);
```

Comment:
Need to align also `parseCtx_Free` to use `rm_free`
It is setting its `ctx->my_id` to what `shardid` returns
(either from rule `shardid ::= STRING` or `shardid ::= INTEGER`)

Reply:
Yes! I had that one changed but it was being marked at the time by the diagnostic as an incorrect rm_free. Presumably because only half of the allocs were rm_. STRINGs were being rm_strdup'd, but INTEGERs were being asprintf'd. I will double check this with the diagnostic.